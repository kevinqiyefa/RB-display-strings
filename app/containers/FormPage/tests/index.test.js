/**
 *
 * Tests for FormPage
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import configureStore from '../../../configureStore';

import { FormPage } from '../index';

describe('<FormPage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const dispatch = jest.fn();
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <FormPage fetchStringRequest={dispatch} />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
