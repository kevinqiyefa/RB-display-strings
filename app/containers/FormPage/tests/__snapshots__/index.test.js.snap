// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormPage /> should render and match the snapshot 1`] = `
<div
  class="ContentWrapper-sc-1pl7b9y-0 kEVOet"
>
  <h1
    class="ContentHeader-cm493b-0 bZemkD"
  >
    Enter a string:
  </h1>
  <form
    class="AddStringForm-sc-13pwdct-0 hOzxvj"
  >
    <div
      class="InputWrapper-sc-133b58z-0 gNAbZM"
    >
      <input
        class="Input-sc-18nxcvk-0 hyFzew"
        placeholder="string"
        type="text"
        value=""
      />
    </div>
    <input
      class="SubmitButton-sc-1t59sxp-0 jCulgx"
      type="submit"
      value="Add"
    />
  </form>
</div>
`;
